### PR TITLE
feat(#9428): collect new size metrics from CHT monitoring API

### PR DESCRIPTION
**Pull request by jkuester**
Issue: https://github.com/medic/cht-core/issues/9428
Docs PR: https://github.com/medic/cht-docs/pull/1545
Core PR: https://github.com/medic/cht-core/pull/9438

### CODE REVIEW

**Comment by jkuester** [comment]
@mrjones-plip one dilemma we have here is that the new fields on the `/monitoring` api will only exist for instances running `4.11+`.  Based on my testing it looks like the `json-exporter` will [log an error](https://github.com/prometheus-community/json_exporter/blob/03abaa7e9c4842bdf8f091e6679f42cd171fd163/exporter/collector.go#L139) for each field in the config that it cannot resolve.  It does not stop execution and will still emit metrics for the other fields, but it will spam its own log file.  I could not find any way to disable this logging or somehow avoid it.  The functionality is still fine, just a bunch of extra logging.

Could perhaps see if the `json-exporter` project would take a PR.  Or maybe we add some logging config for the [json-exporter service](https://github.com/medic/cht-watchdog/blob/0d12c6d6c0e6c358319d65ca4136458341df2a9d/docker-compose.yml#L53) in the compose file to make sure the extra logs will not fill up the disk...  What do you think?

**Review by m5r — APPROVED** [review]
:+1:

File: development/README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Tips:
 
 ### Adding a new metric from CHT /monitoring API
 
-Metrics loaded from the CHT `/monitoring` endpoint are configured via the json-exporter's [`config.yml`](../exporters/json/config/cht.yml) file. The `modules.default.metrics` section contains the configuration for mapping the JSON response from the CHT `/monitoring` endpoint to Prometheus metrics. New values added to this JSON can be included in Prometheus by adding additional mapping here. See the [json_exporter project](https://github.com/prometheus-community/json_exporter) on GitHub for more information.
+Metrics loaded from the CHT `/monitoring` endpoint are configured via the json-exporter's [`config/cht.yml`](../exporters/json/config/cht.yml) file. The `modules.default.metrics` section contains the configuration for mapping the JSON response from the CHT `/monitoring` endpoint to Prometheus metrics. New values added to this JSON can be included in Prometheus by adding additional mapping here. See the [json_exporter project](https://github.com/prometheus-community/json_exporter) on GitHub for more information.
```
[the remark this answers is not in the document]
Comment (jkuester) [reply development/README.md:60]: This is what happens when it has been so long that I have to read my own documentation again to figure out how to contribute to the project.... :sweat_smile:

**Comment by mrjones-plip** [comment]
per our chat: we're OK with the errors as long as we ensure logs are both rotated and compressed (and then discarded) to ensure the health of the container.  thus, we may need to update json exporter upstream, or, more easily, our docker compose in the watchdog repo.

**Comment by jkuester** [comment]
Okay, I have updated all our prod docker services to include a `logging` configuration (the same that we use on [prod cht-core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:master/docker-compose/cht-core.yml)).  They now use the [local logging driver](https://docs.docker.com/engine/logging/drivers/local/) (instead of the default `json` driver).  

And, for the record, the Docker [logging docs](https://docs.docker.com/engine/logging/configure/) do have a big warning at the top saying:

> Tip: use the `local` logging driver to prevent disk-exhaustion
>
> By default, no log-rotation is performed. As a result, log-files stored by the default `json-file` logging driver logging driver can cause a significant amount of disk space to be used for containers that generate much output, which can lead to disk space exhaustion.

File: .env.example [hunk 2]
```diff
@@ -24,6 +24,9 @@ SQL_EXPORTER_IP=127.0.0.1
 # 9399 is default from upstream project
 SQL_EXPORTER_PORT=9399
 
+# Use to customize log retention for docker containers
+#LOG_MAX_SIZE=50m
+#LOG_MAX_FILES=20
```
Comment (m5r) [inline .env.example:29]: FYI to potential future reviewers:
> By default, the local driver preserves 100MB of log messages per container and uses automatic compression to reduce the size on disk. The 100MB default value is based on a 20M default size for each file and a default count of 5 for the number of such files (to account for log rotation).

from https://docs.docker.com/engine/logging/drivers/local/

**Pull request merged**